pyro/autohelp/ast_visitor.py

```diff
@@ -0,0 +1,246 @@
+import enum
+from typing import Optional
+
+import libcst
+
+
+class Actions(enum.Enum):
+    # we don't care about any of these values, just using them as sentinels
+    DECORATOR_EVENT_CALLED = enum.auto()
+    DECORATOR_LISTEN_NOT_CALLED = enum.auto()
+    USING_SELF_ON_BOT_COMMAND = enum.auto()
+    CLIENT_IS_NOT_BOT = enum.auto()
+    PROCESS_COMMANDS_NOT_CALLED = enum.auto()
+
+
+class BaseHelpTransformer(libcst.CSTTransformer):
+    def __init__(self, find_all: bool = False):
+        super().__init__()
+        self.found_errors: list[Actions] = []
+        self.find_all = find_all
+        self.updates = {}
+
+    def _check_code(self) -> bool:
```

Comment:
Could this be a property?

Reply:
Didn't even use this anywhere in the final implementation, removed in https://github.com/MenuDocs/Pyro/pull/58/commits/34b7e4fbc198398a3c57295f020a8c1420dd10bc